src/amenities/amenities_server/controller/AmenitiesController.cs

```diff
@@ -0,0 +1,104 @@
+using Microsoft.AspNetCore.Mvc;
+
+namespace amenities_server.Controllers;
+
+[ApiController]
+[Route("amenities")]
+public class AmenitiesController : ControllerBase
+{
+    private IAmenityPersistence _amenityPersistence;
+
+    public AmenitiesController()
+    {
+        _amenityPersistence = Services.GetAmenityPersistence();
+    }
+
+    //get: /amenities
+    [HttpGet]
+    public IActionResult GetAmenities()
+    {
+        var amenities = _amenityPersistence.GetAmenities();
+
+        if (amenities == null)
+        {
+            return NotFound();
+        }
+
+        return Ok(new AmenityResponse<IEnumerable<Amenity>>("Amenities retrieved successfully.", amenities));
+    }
+
+    //get: /amenities/{id}
+    [HttpGet("{id}")]
+    public IActionResult GetAmenityByID(int id)
+    {
+        var amenity = _amenityPersistence.GetAmenityByID(id);
+
+        if (amenity == null)
+        {
+            return NotFound(new AmenityResponse<int>("Amenity with specified id not found.", id));
+        }
+
+        return Ok(new AmenityResponse<Amenity>("Amenity retrieved successfully.", amenity));
+    }
+
+    //delete: /amenities/{id}
+    [HttpDelete("{id}")]
+    public IActionResult DeleteAmenity(int id)
+    {
+        //TODO:  validate session call
+        //if(!Services.GetPermissionValidator().ValidatePermissions(permission,sessionKey))
+
+        //validate passed amenity
+        var amenity = _amenityPersistence.GetAmenityByID(id);
+        if(amenity == null){
+            return BadRequest(new AmenityResponse<int>("Bad Request. Amenity with specified id not found.", id));
+        }
+
+        _amenityPersistence.DeleteAmenity(id);
+
+        return Ok(new AmenityResponse<string>("Amenity deleted successfully.", null));
+    }
+    //post: /amenities
+    [HttpPost]
+    public IActionResult AddAmenity(Amenity newAmenity)
+    {
+        //TODO: validate session call
+        //if(!Services.GetPermissionValidator().ValidatePermissions(permission,sessionKey))
+
+        //validate passed amenity
+        if (!AmenityValidator.ValidateNewAmenity(newAmenity))
```

Comment:
__Suggestion only__

An alternative to a public static validator is to use model validation. 
That would require modifying the Amenity model to use [System.ComponentModel.DataAnnotations](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations) with annotation like [required]  before critical fields. Then add the [ValidateModel] annotation to any controllers that need it.

[This document](https://learn.microsoft.com/en-us/aspnet/web-api/overview/formats-and-model-binding/model-validation-in-aspnet-web-api) give a nice overview of model validation.

Reply:
great suggestion, I want to implement this but in later sprints as we don't have time now